fil_fungible_token/src/token/mod.rs

```diff
@@ -535,6 +548,29 @@ mod test {
         assert_eq!(last_called, expected);
     }
 
+    #[test]
+    fn it_wraps_a_previously_loaded_state_tree() {
+        struct ActorState {
+            token_state: TokenState,
+        }
+
+        // simulate the token state being a node in a larger state tree
+        let bs = MemoryBlockstore::default();
+        let actor_state = ActorState { token_state: TokenState::new(&bs).unwrap() };
+
+        // wrap the token state, moving it into a TokenHandle
+        let mut token = Token::wrap(
+            &bs,
+            FakeMessenger::new(TOKEN_ACTOR.id().unwrap(), 6),
+            actor_state.token_state,
+        );
+
+        token.mint(TOKEN_ACTOR, TREASURY, &TokenAmount::from(1), &Default::default()).unwrap();
+        // gets a read-only state
+        let state = token.borrow_state();
```

Comment:
Hmm, but what about `actor_state.token_state`. Did `RefCell::new` copy the state (it implements Clone, but not Copy...)? `Wrap` wouldn't be the right word to describe what's happening if so, and I see decent potential for error here inspecting the old state. Did the old one get moved so it can't be read any more?

Could we explore more what's the deal with the two states now? Included commented-out lines for things that don't compile, to explain that.

We don't need to eliminate clones at all costs, but would like a justification for them.

Reply:
`RefCell::new` will take ownership of the state so it's da move (memcpy) rather than a clone. 

`actor_state.token_state` is not valid after line 565 anymore.

Reply:
Ok, even memcpy is a bit undesirable, but at least it's safe.

Reply:
Can a different field in actor_state be accessed? If the partial move invalidates the whole containing struct, it will be awkward for uses that have embedded token state.

Reply:
Yeh, the entire actor_state becomes invalid after the partial move so this is pretty awkward compared to the &mut impl